src/main/java/am/ik/yavi/core/Constraint.java

```diff
@@ -33,6 +38,18 @@ default C isNull() {
 		return this.cast();
 	}
 
+	default C equalTo(@Nullable V other) {
+		this.predicates().add(ConstraintPredicate.of(Predicate.isEqual(other), OBJECT_EQUAL_TO,
+				() -> new Object[] { other }, NullAs.INVALID));
+		return this.cast();
+	}
+
+	default C oneOf(Collection<V> values) {
+		this.predicates().add(ConstraintPredicate.of(values::contains, OBJECT_ONE_OF,
+				() -> new Object[] { values.toArray() }, NullAs.INVALID));
```

Comment:
It is not an absolute requirement, you can pass the arguments as collection as well. Although I would always make a copy first to prevent inconsistent failure messages if the collection is changed after validation.

Reply:
The string representation of an array is just useless, isn't it?

Just `new Object[] { values }` should be fine.

Reply:
or `new Object[] { Arrays.toString(values.toArray()) }` if you want to make a copy

Reply:
You're right about the useless array string representation. 
Imho `new Object[] { new ArrayList<>(values) }` is the best way to proceed:

1. This way you get an understandable string representation.
2. No inconsistencies happen because of the defensive copy.
3. Users can process the arguments their own way of they want to, without having to parse a string representation.

Reply:
Looks good 👍